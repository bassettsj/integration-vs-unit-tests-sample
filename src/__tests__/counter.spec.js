// ./__tests__/counter.spec.js
import { shallowMount } from '@vue/test-utils';
import { render, fireEvent, cleanup } from '@testing-library/vue';
import Counter from '../counter.vue';
import CounterButton from '../counter-button.vue';

describe('Counter component', () => {
  describe('Unit Tests', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallowMount(Counter);
    });

    it('should render', () => {
      // Doesn't catch the real snapshot.
      expect(wrapper).toMatchSnapshot();
    });

    it('should increment when button is clicked', () => {
      // Passes but it is not actually working together.
      wrapper.find(CounterButton).vm.$emit('click');
      expect(wrapper.vm.count).toBe(1);
    });
  });
  describe('Integration Tests', () => {
    afterEach(cleanup);
    it('should increment when you click the + button', async () => {
      const { getByText } = render(Counter);
      getByText('Times clicked: 0');
      const increment = getByText('+');
      await fireEvent.click(increment);
      getByText('Times clicked: 1');
    });

    it('should decrement when you click the - button', async () => {
      const { getByText } = render(Counter);
      getByText('Times clicked: 0');
      const increment = getByText('+');
      const decrement = getByText('-');
      await fireEvent.click(increment);
      getByText('Times clicked: 1');
      await fireEvent.click(decrement);
      getByText('Times clicked: 0');
    });
  });
});
