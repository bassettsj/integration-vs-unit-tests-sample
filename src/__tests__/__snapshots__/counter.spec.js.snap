// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component Unit Tests should render 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
